src/bridge/cockpitcpusamples.c

```diff
@@ -112,3 +112,72 @@ cockpit_cpu_samples (CockpitSamples *samples)
   g_strfreev (lines);
   g_free (contents);
 }
+
+void
+cockpit_cpu_temperature (CockpitSamples *samples)
+{
+  for (int i = 0; TRUE; i++)
+    {
+      gsize len;
+      g_autoptr (GError) error = NULL;
+      g_autofree gchar *path = g_strdup_printf ("/sys/class/hwmon/hwmon%d/name", i);
+      g_autofree gchar *name = NULL;
+
+      if (!g_file_get_contents (path, &name, &len, &error))
```

Comment:
This pattern of "read the file, and ignore it if it doesn't exist, otherwise abort with an error" is repeated quite a lot.  It probably deserves a helper function of some kind.